crates/sequencer/src/rpc.rs

```diff
@@ -334,6 +347,70 @@ impl SequencerRpcServer for SequencerRpcServerImpl {
                 internal_rpc_error(format!("Could not send resume commitments signal: {e}"))
             })
     }
+
+    /// Subscribe to Citrea events
+    async fn subscribe_citrea(
+        &self,
+        pending: PendingSubscriptionSink,
+        topic: String,
+    ) -> SubscriptionResult {
+        match topic.as_str() {
+            "newL2Blocks" => {
+                let subscription = pending.accept().await?;
+                let mut rx = self.context.l2_block_rx.resubscribe();
+                let storage = self.context.storage.clone();
+                let ledger = self.context.ledger.clone();
+
+                tokio::spawn(async move {
+                    while let Ok(block_height) = rx.recv().await {
```

Comment:
No, broadcast::Receiver::recv() can return a meaningful error:

```
enum RecvError {
    Closed,
    Lagged(u64),
}
```

It case of lagged we may actually retry .recv(), see how much we lagged and push lagged blocks along with the new fresh block.